pkg/controller/operators/olm/operator.go

```diff
@@ -521,19 +624,55 @@ func (a *Operator) crdOwnerConflicts(in *v1alpha1.ClusterServiceVersion, csvsInN
 	return nil
 }
 
-// annotateNamespace is the method that gets called when we see a namespace event in the cluster
-func (a *Operator) annotateNamespace(obj interface{}) (syncError error) {
+// syncNamespace is the method that gets called when we see a namespace event in the cluster
+func (a *Operator) syncNamespace(obj interface{}) (syncError error) {
 	namespace, ok := obj.(*corev1.Namespace)
+	namespaceName := namespace.GetName()
 	if !ok {
 		log.Debugf("wrong type: %#v", obj)
 		return fmt.Errorf("casting Namespace failed")
 	}
 
-	log.Infof("syncing Namespace: %s", namespace.GetName())
+	log.Infof("syncing Namespace: %s", namespaceName)
 	if err := a.annotator.AnnotateNamespace(namespace); err != nil {
-		log.Infof("error annotating namespace '%s'", namespace.GetName())
+		log.Infof("error annotating namespace '%s'", namespaceName)
+		return err
+	}
+
+	opGroupList, err := a.client.OperatorsV1alpha2().OperatorGroups(namespaceName).List(metav1.ListOptions{})
```

Comment:
You can just query the operatorgroup informer here instead. @njhale can point you to an example in the package server :)

Reply:
@jpeeler You can get a `Lister` from the `OperatorGroup` informer that lets you query the indexer's cache instead of making an API call. As an example, we [do this for checking RBAC resources](https://github.com/operator-framework/operator-lifecycle-manager/blob/177560083fafc69c3ec94444b2a6370f953a2feb/pkg/controller/operators/olm/operator.go#L122).